### PR TITLE
Some fixes for line numbering, and fixes for license error handling

### PR DESCRIPTION
**Pull request by AlanCoding**
##### ISSUE TYPE

 - Feature Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME

 - API
 - UI
 - Installer

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by AlanCoding** [comment]
This still seems to hit:

```
2020-11-17 14:39:11,379 ERROR    awx.main.tasks inventory_update 3 (running) Post run hook errored.
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/tasks.py", line 1530, in run
    self.post_run_hook(self.instance, status)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/tasks.py", line 2802, in post_run_hook
    cmd.perform_update(options, data, inventory_update)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 1035, in perform_update
    raise e
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 1032, in perform_update
    self.check_license()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 853, in check_license
    raise CommandError('No license found!')
django.core.management.base.CommandError: No license found!
```

What we don't want is for `post_run_hook` to error with an arbitrary exception (like CommandError).

This was supposed to be converted into the new exception type to be handled by the main `run` method, but I don't know why it was not.

File: awx/main/tasks.py [hunk 1]
```diff
@@ -1534,6 +1528,12 @@ def run(self, pk, **kwargs):
 
         try:
             self.post_run_hook(self.instance, status)
+        except PostRunError as exc:
+            if status == 'successful':
+                status = exc.status
+                extra_update_fields['job_explanation'] = exc.args[0]
+                if exc.tb:
+                    extra_update_fields['result_traceback'] = exc.tb
```
Comment (AlanCoding) [inline awx/main/tasks.py:1536]: @jladdjr I wanted to draw your attention to this.

I don't like modifying the logic inside of the main `run` method, but I can't think of any other way to do this which won't flip the status of the update back-and-fourth. In these cases, the `ansible-inventory` subprocess is fully successful, but for some reason, the saving process errors.

File: awx/main/tasks.py [hunk 2]
```diff
@@ -2790,42 +2790,19 @@ def emit(self, record):
             job_timeout=self.get_instance_timeout(self.instance),
             counter=self.event_ct
         )
+        handler._start_line = self.end_line
```
Comment (AlanCoding) [inline awx/main/tasks.py]: this was kind of lazy, we could make the starting line a kwarg to the handler class

**Review by jladdjr — APPROVED** [review]

**Comment by jladdjr** [comment]
> Only those with write access to this repository can merge pull requests.

feel free to merge, looks like I can't do that for things pointed at your repo

**Pull request merged**